tests/unittests/unittests.cpp

```diff
@@ -4264,6 +4291,96 @@ class ConcurrentQueueTests : public TestClass<ConcurrentQueueTests>
 		
 		return true;
 	}
+
+	bool acquire_and_signal()
+	{
+        typedef ::moodycamel::LightweightSemaphore semaphore;
+
+		{
+            semaphore s;
+			s.signal();
+
+			std::thread t1(threadTestOne, std::ref(s));
```

Comment:
I think `threadTestOne` and `threadTestN` would be better off as local lambda variables, since their usage is highly dependent on this particular unit test.